### PR TITLE
Mark the library as typed, plus a couple of other tweaks

### PR DESCRIPTION
**Pull request by davep**
Mostly this aims to mark the library as fully typed, so that type checkers pick up on all of the type hinting that's already in the library (thank you!). Also seeks to improve one type within the code, and also simplifies the core exception class.

### CODE REVIEW

**Comment by Damego** [comment]
Wow. I didn't expect any pr's for this library because it's have very few downloads

**Comment by Damego** [comment]
(Just looked at code of my library and im like who wrote this shit lmao)

File: bored_api/bored_api.py [hunk 1]
```diff
@@ -36,11 +34,11 @@ class BoredActivity:
 class BoredClient:
     BASE_URL = "http://www.boredapi.com/api/activity/?"
 
-    async def _get_request(self, url: str) -> dict:
+    async def _get_request(self, url: str) -> dict[str,Any]:
```
Comment (Damego) [inline bored_api/bored_api.py]:
```suggestion
    async def _get_request(self, url: str) -> dict[str, Any]:
```

File: bored_api/bored_api.py [hunk 2]
```diff
@@ -36,11 +34,11 @@ class BoredActivity:
...
         """
         Makes a GET request to bored api
         :param url: The url.
-        :return: dict
+        :return: dict[str,Any]
```
Comment (Damego) [inline bored_api/bored_api.py]:
```suggestion
        :return: dict[str, Any]
```

**Review by Damego — COMMENTED** [review]
One moment

**Comment by davep** [comment]
> Wow. I didn't expect any pr's for this library because it's have very few downloads

Finding it very handy, I've been using it as the core of [a test app I'm building with Textual](https://github.com/davep/unbored). Been having a lot of fun with it. :-)

https://www.youtube.com/watch?v=4rOIngibPhw

**Comment by Damego** [comment]
> > Wow. I didn't expect any pr's for this library because it's have very few downloads
> 
> Finding it very handy, I've been using it as the core of [a test app I'm building with Textual](https://github.com/davep/unbored). Been having a lot of fun with it. :-)
> 
> https://www.youtube.com/watch?v=4rOIngibPhw

Cool!

**Review by Damego — APPROVED** [review]
Thanks for contributing!

**Comment by davep** [comment]
>Thanks for contributing!

My pleasure. Thanks for making this available!

**Pull request merged**